test/wallet.test.ts

```diff
@@ -156,6 +179,32 @@ describe('wallet', () => {
         );
       }
     });
+
+    it('should throw when provided unknown address', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice(0, 2);
+      const witnessedTxParams: TransactionParams[] = [];
+      const processSignTransaction = async (_txParams: TransactionParams) => {
+        witnessedTxParams.push(_txParams);
+        return testTxHash;
+      };
+
+      engine.push(
+        createWalletMiddleware({ getAccounts, processSignTransaction }),
+      );
+      const txParams = {
+        from: testAddresses[0],
+      };
+
+      const payload = { method: 'eth_signTransaction', params: [txParams] };
+      try {
+        await pify(engine.handle).call(engine, payload);
+      } catch (e: any) {
+        expect(e.message).toEqual(
+          'The requested account and/or method has not been authorized by the user.',
+        );
+      }
```

Comment:
This test looks very, very similar to the happy path above. Are you sure this is throwing an error? P.S. it's better to write this as:

``` suggestion
      const promise = pify(engine.handle).call(engine, payload);
      await expect(promise).rejects.toThrow('The requested account and/or method has not been authorized by the user.')
```